README.md

```diff
@@ -349,6 +349,13 @@ conn
 |> assign_prop(:paginated_list, inertia_defer(&calculate_next_page/0) |> inertia_merge())
 ```
 
+If you are working with complex data structures or nested objects you can use `inertia_deep_merge(value)`
```

Comment:
@derrickreimer I contemplated just adding the support to the `inertia_merge` as something like:
```elixir
inertia_merge(value, :deep)
```

But I thought that this could bring some confusion and it's less explicit that the proposed solution.